README.md

```diff
@@ -55,3 +55,72 @@ See `k8s-jenkins-cac.tf` in your project.
+
+An "environment" is the set of machines where your microservices will run. For example: you will probably have some environments: "development", "staging" and "production". In a cloud provider like GCP, we recommend separating the environments on a project level. This means that one gcp project correlates to one kuberpult environment 1:1 - although this is not a technical requirement.
+
+Environments are also called "stages" - but in kuberpult we stick to "environments", short "envs".
+
+The config for an environment is stored in a json file called `config.json`. This file would go into an environment in the manifests repository like this: `environments/development/config.json` (in this example the `config.json` file would dictate the configuration for the `development` environment)
+
+In the `config.json` file there are 3 main fields:
+- [Upstream](#upstream)  `"upstream"`
+- [ArgoCD](#argocd)    `"argocd"`
+- [EnvironmentGroup](#environment-group) `"environmentGroup"`
+
+##### Upstream:
+
+The `"upstream"` field can have one of the two options (cannot have both):
+  - `latest`: can only be set to `true`
+  - `environment`: has a string which is the name of another environment. Following the chain of upstream environments would take you to the one with `"latest": true`
+
+##### ArgoCd: 
+
+The `"argocd"` field has a few subfields:
+- `"accessList"`:  
+  Is a list of objects that describe which resources are allowed in the cluster ([ArgoCD Docs](https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#projects))
+  It has the following fields:
+  - `"group"`: relates to the Kubernetes API group without the version
+  - `"kind"`: the resource type in kubernetes
+
+- `"destination"` (**Mandatory**):
+  > Defines which Kubernetes cluster/namespace to deploy to
+  >
+  > \- [Template ArgoCD Docs](https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Template/#template-fields)
+
+  It has the following fields:
+  - `"name"` (**Mandatory**): Name of the cluster (within Argo CD) to deploy to
+  - `"server"` (**Mandatory**): API Server URL for the cluster (Example: `"https://kubernetes.default.svc"`)
+  - `"namespace"`: Target namespace in which to deploy the manifests from source (Example: `"my-app-namespace"`)
+
+- `"syncWindows"`:
+  > Sync windows are configurable windows of time where syncs will either be blocked or allowed
+  >
+  > - [Sync Windows ArgoCD Docs](https://argo-cd.readthedocs.io/en/stable/user-guide/sync_windows/)
+
+  It has the following fields:
+  - `"schedule"`: is the schedule of the window in `cron` format (Example: `"10 1 * * *"`)
+  - `"duration"`: the duration of the window (Example: `"1h"`)
+  - `"kind"`: defines whether the sync is allowed (`"allow"`) or denied (`"deny"`)
+  - `"applications"`: an array with the names of the applications
+
+- `"ignoreDifferences"`:
+  > Argo CD allows ignoring differences at a specific JSON path, using RFC6902 JSON patches and JQ path expressions.
```

Comment:
please give an example why this is necessary. e.g.
```suggestion
  > Argo CD allows ignoring differences at a specific JSON path, using RFC6902 JSON patches and JQ path expressions.
  This is useful, because some kubernetes resources have fields that change fields over time (e.g. field `todo.field.name.here`). By default ArgoCd will sync everything, so you may need to exclude these fields. 
```

Reply:
just saw that you have an example further below, so it's not necessary here.

Reply:
I added the detail about 'the fields that change overtime' anyway, I think it's a nice addition!